backend/fms_core/services/library.py

```diff
@@ -0,0 +1,118 @@
+from django.core.exceptions import ValidationError
+
+from fms_core.models import LibraryType, Library
+
+from fms_core.services.process import create_process
+from fms_core.services.container import create_container, get_container
+from fms_core.services.sample import transfer_sample, remove_qc_flags
+from fms_core.services.property_value import create_process_properties
+
+from datetime import datetime
+
+
+def get_library_type(name):
+    instrument = None
+    errors = []
+    warnings = []
+    try:
+        library_type = LibraryType.objects.get(name=name)
+    except LibraryType.DoesNotExist as e:
+        errors.append(f"No Experiment Type named {name} could be found.")
+    except LibraryType.MultipleObjectsReturned as e:
+        errors.append(f"More than one Experiment Type was found with the name {name}.")
+
+    return instrument, errors, warnings
+
+def create_library(library_type, library_size, index, platform):
+    library = None
+    errors = []
+    warnings = []
+
+    if not all([library_type, index, platform]):
+        errors.append('Experiment type, Index and Platform are required to create a library.')
+        return library, errors, warnings
+
+    try:
+        library = Library.objects.create(library_type=library_type,
+                                         library_size=library_size,
+                                         index=index,
+                                         platform=platform)
+    except ValidationError as e:
+        errors.append(';'.join(e.messages))
+
+    return library, errors, warnings
+
+
+
+def prepare_library(library_type, library_size, platform, library_date,
+                    library_rows_info, protocol, process_properties, comment):
+    library = None
+    errors = []
+    warnings = []
+
+    if not library_type:
+        errors['library_type'] = "Invalid library type"
+        return library, errors, warnings
+
+    process, process_errors, process_warnings = \
+        create_process(protocol=protocol,
+                       creation_comment=comment if comment else f"Automatically generated via library preparation "
+                                                                f"on {datetime.utcnow().isoformat()}Z",)
+
+    # Create process' properties
+    if not process_errors:
+        properties, properties_errors, properties_warnings = create_process_properties(process_properties,
+                                                                                       process)
+
+    errors += process_errors + properties_errors
```

Comment:
tabulation here and next line. this will fail if there is process_errors (properties_errors won't be initialized and "+" won't be defined for NoneType)